### PR TITLE
jak1: clean up all `dummy` methods

### PR DESCRIPTION
**Pull request by Hat-Kid**
Cleans up every `dummy-*` and `TODO-RENAME-*` method up with either proper names or by renaming them to `[type-name]-method-[method-id]` similar to Jak 2's `all-types`.

Also fixes the bad format string in `collide-cache` and adds the event handler hack to Jak 1.

The game boots and runs fine, but I might have missed a PAL patch or other manual patches here and there, please double-check if possible.

### CODE REVIEW

File: goal_src/jak1/levels/jungle/fisher.gc [hunk 1]
```diff
@@ -1687,18 +1687,12 @@
       )
     (none)
     )
-  :exit (behavior ()
-    (#when PC_PORT
-    ;; needed for cutscene skipping
-      (set! (-> self cur-trans-hook) nothing)
-      )
-    ((-> (method-of-type process-taskable play-anim) exit)))
```
Comment (ManDude) [inline goal_src/jak1/levels/jungle/fisher.gc:1695]: Also here.

File: goal_src/jak1/levels/maincave/baby-spider.gc [hunk 2]
```diff
@@ -265,8 +265,8 @@ baby-spider-default-event-handler
   (none)
   )
 
-(defmethod dummy-52 baby-spider ((obj baby-spider) (arg0 vector))
-  (+! (-> obj wiggle-angle) (* DISPLAY_FPS_RATIO (-> obj delta-wiggle-angle))) ;; changed for high fps
+(defmethod nav-enemy-method-52 baby-spider ((obj baby-spider) (arg0 vector))
+  (+! (-> obj wiggle-angle) (-> obj delta-wiggle-angle))
```
Comment (ManDude) [inline goal_src/jak1/levels/maincave/baby-spider.gc:269]: Same thing here.

File: goal_src/jak1/levels/title/title-obs.gc [hunk 3]
```diff
@@ -490,16 +488,15 @@
...
   :trans (behavior ()
     ((-> (method-of-type logo startup) trans))
-    (if  (and (or *debug-segment* PC_PORT)
-              (cpad-pressed? 0 start circle x)
-              (member (level-status *level* 'village1) '(loaded active))
-              )
+    (if (and *debug-segment*
+             (cpad-pressed? 0 start circle x)
+             (member (level-status *level* 'village1) '(loaded active))
+             )
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:494]: Patch here.

File: goal_src/jak1/levels/title/title-obs.gc [hunk 4]
```diff
@@ -490,16 +488,15 @@
...
         (set-setting! 'bg-a 'abs 1.0 0)
         )
     (when (and (= (-> *setting-control* current bg-a) 1.0)
                (and (member (level-status *level* 'village1) '(active loaded))
-                    ;; i think we can forgo the time check here...
-                    (#if PC_PORT #t (>= (- (-> *display* base-frame-counter) (-> self state-time)) (seconds 3)))
+                    (>= (- (-> *display* base-frame-counter) (-> self state-time)) (seconds 3))
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:499]: And here.

File: goal_src/jak1/levels/title/title-obs.gc [hunk 5]
```diff
@@ -557,27 +554,12 @@
...
-          (the-as art-joint-anim #f)
-          (the-as (function process-drawable symbol) false-func)
-          )
-        ))
+    (ja-play-spooled-anim
+      (-> self anim)
+      (the-as art-joint-anim #f)
+      (the-as art-joint-anim #f)
+      (the-as (function process-drawable symbol) false-func)
+      )
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:562]: And here.

File: goal_src/jak1/levels/title/title-obs.gc [hunk 6]
```diff
@@ -599,19 +581,12 @@
...
-           )
-         (else
-           (initialize-skeleton self *logo-sg* '())
-           )
+     (case (scf-get-territory)
+       ((2)
+        (initialize-skeleton self *logo-japan-sg* '())
+        )
+       (else
+         (initialize-skeleton self *logo-sg* '())
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:589]: Here.

File: goal_src/jak1/levels/title/title-obs.gc [hunk 7]
```diff
@@ -707,18 +682,13 @@
   :trans (behavior ()
     (hide-hud-quick)
     (spool-push *art-control* "ndi-intro" 0 self -1.0)
-    ;; PAL patch here
-    (if *sound-player-enable*
-        (sound-group-pause (sound-group music))
-        )
+    (sound-group-pause (sound-group music))
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:685]: Here

File: goal_src/jak1/levels/title/title-obs.gc [hunk 8]
```diff
@@ -707,18 +682,13 @@
...
     (none)
     )
   :code (behavior ()
     (let ((gp-0 (the-as handle #f)))
-      ;; PAL patch here
-      (cond
-        ((and (= (scf-get-territory) GAME_TERRITORY_SCEI) *first-boot*)
-         (set! *first-boot* #f)
+      (case (scf-get-territory)
+        ((2)
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:691]: Here

File: goal_src/jak1/levels/title/title-obs.gc [hunk 9]
```diff
@@ -855,7 +825,7 @@
     (remove-setting! 'ambient-volume)
     (remove-setting! 'sfx-volume)
     (remove-setting! 'music-volume)
-    ;; PAL patch here (code removal)
+    (remove-setting! 'allow-progress)
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:828]: Here.

File: goal_src/jak1/levels/title/title-obs.gc [hunk 10]
```diff
@@ -868,22 +838,15 @@
       (set! (-> *time-of-day-proc* 0 time-ratio) 300.0)
       (set! *time-of-day-fast* #f)
       )
-    ;; PAL patch here
     (remove-setting! 'allow-pause)
-    (remove-setting! 'allow-progress)
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:873]: Here

File: goal_src/jak1/levels/title/title-obs.gc [hunk 11]
```diff
@@ -868,22 +838,15 @@
...
-    (when (cpad-pressed? 0 start)
-      (set-setting! 'allow-progress #t 0.0 0)
-      (apply-settings *setting-control*)
-      (activate-progress *dproc* (progress-screen title))
-      (set-setting! 'allow-progress #f 0.0 0)
-      (apply-settings *setting-control*)
-      )
+    (if (cpad-pressed? 0 start)
+        (activate-progress *dproc* (progress-screen title))
+        )
```
Comment (ManDude) [inline goal_src/jak1/levels/title/title-obs.gc:849]: Here

File: goal_src/jak1/levels/village1/yakow.gc [hunk 12]
```diff
@@ -204,27 +202,12 @@ yakow-default-event-handler
...
-                )
-              (none)
-              )
-            :to self
+          (level-hint-spawn
+            (game-text-id village1-yakow-resolution)
+            "sksp018a"
+            (the-as entity #f)
+            *entity-pool*
+            (game-task none)
```
Comment (ManDude) [inline goal_src/jak1/levels/village1/yakow.gc:210]: Here

**Review by ManDude — CHANGES_REQUESTED** [review]

File: goal_src/jak1/engine/collide/collide-edge-grab.gc [hunk 13]
```diff
@@ -730,90 +731,73 @@
...
-   )
+  (set! *no-walk-surface* v1-2)
+  (set! (-> v1-2 mult-hook) (the-as (function surface surface surface int none) nothing))
   (set! (-> v1-2 touch-hook) nothing)
   (set! (-> v1-2 active-hook) nothing)
   )
-
-
-; (defmethod find-edge-grabs! target ((obj target) (arg0 collide-cache))
-;   (format *stdcon* "no edge grabs for you!~%")
```
Comment (water111) [inline goal_src/jak1/engine/collide/collide-edge-grab.gc:817]: wow, I guess this has been here for a while...

File: goal_src/jak1/engine/collide/collide-shape.gc [hunk 14]
```diff
@@ -2426,18 +2427,17 @@
...
+    (cond
+      ((and (zero? (-> obj prim-core collide-as)) (zero? (-> obj collide-with)))
+       (new 'static 'rgba :r #x80 :g #x80 :b #x80 :a #x40)
+       )
+      ((logtest? (-> obj prim-core action) (collide-action solid))
+       (new 'static 'rgba :r #xff :g #xff :a #x40)
+       )
+      (else
+        (new 'static 'rgba :r #xff :g #x80 :a #x40)
+        )
```
Comment (water111) [inline goal_src/jak1/engine/collide/collide-shape.gc:2439]: cool to see that this works now

File: goal_src/jak1/engine/common-obs/rigid-body.gc [hunk 15]
```diff
@@ -451,13 +451,13 @@
   (+! (-> obj sim-time-remaining)
       (* 0.0033333334 (the float (- (-> *display* base-frame-counter) (-> *display* old-base-frame-counter))))
       )
-  (let ((f30-0 (* DISPLAY_FPS_RATIO 0.016666668)) ;; changed for high fps
+  (let ((f30-0 0.016666668)
```
Comment (water111) [inline goal_src/jak1/engine/common-obs/rigid-body.gc:454]: Missing patch

File: goal_src/jak1/engine/debug/stats-h.gc [hunk 16]
```diff
@@ -120,14 +121,14 @@
 
 (defmethod update-wait-stats perf-stat ((obj perf-stat) (arg0 uint) (arg1 uint) (arg2 uint))
   (when (nonzero? (-> obj ctrl))
-   (set! (-> obj to-vu0-waits) (+ (-> obj to-vu0-waits) arg0))
-   (set! (-> obj to-spr-waits) (+ (-> obj to-spr-waits) arg1))
-   (set! (-> obj from-spr-waits) (+ (-> obj from-spr-waits) arg2))
-   )
+    (+! (-> obj to-vu0-waits) arg0)
+    (+! (-> obj to-spr-waits) arg1)
+    (+! (-> obj from-spr-waits) arg2)
```
Comment (water111) [inline goal_src/jak1/engine/debug/stats-h.gc:126]: wow, this must be ancient if `+!` wasn't recognized!

**Review by water111 — APPROVED** [review]
Thanks for doing the cleanup! I just found one missing thing, but I'll grab it in a separate PR.

**Pull request merged**